src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -837,56 +842,181 @@ def create_dns_record_set(resource_group_name, zone_name, record_set_name, recor
     return ncf.create_or_update(resource_group_name, zone_name, record_set_name,
                                 record_set_type, record_set)
 
+def export_zone(resource_group_name, zone_name, file_name):
+    client = get_mgmt_service_client(DnsManagementClient)
+    record_sets = client.record_sets.list_all_in_resource_group(resource_group_name, zone_name)
+
+    record_property_types = {
+        'arecords': 'a',
+        'aaaa_records': 'aaaa',
+        'cname_record': 'cname',
+        'mx_records': 'mx',
+        'ns_records': 'ns',
+        'ptr_records': 'ptr',
+        'soa_record': 'soa',
+        'srv_records': 'srv',
+        'txt_records': 'txt'
+        }
+
+    zone_obj = {
+        '$origin': zone_name.rstrip('.') + '.'
+        }
+
+    for record_set in record_sets:
+        for property_name, record_type in record_property_types.items():
+            record_data = getattr(record_set, property_name, None)
+            if not record_data:
+                continue
+            if not isinstance(record_data, list):
+                record_data = [record_data]
+            if record_type not in zone_obj:
+                zone_obj[record_type] = []
+            for record in record_data:
+                record_obj = {'ip': record.ipv6_address} if record_type == 'aaaa' \
```

Comment:
pylint won't let me do elif on a single line, so this would be twice the number of lines.  I could do a dict with get(,None)s, but I opted for this approach to avoid the extra statements on every field

---
In reply to: [77867357](https://github.com/Azure/azure-cli/pull/826#discussion_r77867357) [](ancestors = 77867357)